src/figuresClasses.ts

```diff
@@ -1,11 +1,84 @@
-export interface Figure {}
+export type Shape = `triangle` | `circle` | `rectangle`;
+export type Color = `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+function roundDown(number: number): number {
+  return Math.floor(number * 100) / 100;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = `triangle`;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides of the triangle must be greater than 0.');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('The sum of any 2 sides must be greater than 3rd side.');
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+    const semiP = (a + b + c) / 2;
```

Comment:
```suggestion
    const semiPerimeter = (a + b + c) / 2;
```
Don't shorten variables' names, names should be descriptive